packages/core/src/oidc-utils.ts

```diff
@@ -0,0 +1,119 @@
+import * as actions_http_client from '@actions/http-client'
+import {IHeaders,IRequestOptions} from '@actions/http-client/interfaces'
+import {HttpClient} from '@actions/http-client'
+import {BearerCredentialHandler} from '@actions/http-client/auth'
+import {debug} from './core'
+
+interface IOidcClient {
+
+  createHttpClient(): actions_http_client.HttpClient
+
+  getApiVersion(): string
+
+  getRuntimeToken(): string
+
+  getIDTokenUrl(): string
+
+  isSuccessStatusCode(statusCode?: number): boolean
+
+  postCall(id_token_url: string, audience: string): Promise<string>
+
+  parseJson(body: string): string
+
+  getIDToken(audience: string): Promise<string>
+}
+
+export class OidcClient implements IOidcClient {
+
+  createHttpClient(allowRetry = true, maxRetry = 10) {
+    let requestOptions : IRequestOptions = {}
+    requestOptions.allowRetries = allowRetry
+    requestOptions.maxRetries = maxRetry
+    return new HttpClient('actions/oidc-client', [
+      new BearerCredentialHandler(this.getRuntimeToken())],
+      requestOptions)
+  }
+
+  getApiVersion(): string {
+    return '2.0'
+  }
+
+  getRuntimeToken(){
+    const token = process.env['ACTIONS_RUNTIME_TOKEN']
+    if (!token) {
+      throw new Error('Unable to get ACTIONS_RUNTIME_TOKEN env variable')
+    }
+    return token
+  }
+
+  getIDTokenUrl(){
+    let runtimeUrl = process.env['ACTIONS_ID_TOKEN_REQUEST_URL']
+    if (!runtimeUrl) {
+      throw new Error('Unable to get ACTIONS_ID_TOKEN_REQUEST_URL env variable')
+    }
+    return runtimeUrl + '?api-version=' + this.getApiVersion()
+  }
+
+  isSuccessStatusCode(statusCode?: number): boolean {
+    if (!statusCode) {
+      return false
+    }
+    return statusCode >= 200 && statusCode < 300
+  }
+
+  async postCall(id_token_url: string, audience: string): Promise<string> {
+
+    const httpclient = this.createHttpClient()
+    if (httpclient === undefined) {
+      throw new Error(`Failed to get Httpclient `)
+    }
+
+    debug(`Httpclient created ${httpclient} `) // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
```

Comment:
what's the output looks like for this trace? will it dump the entire client object? what will that help for debugging?

Reply:
i checked and it showed "object" string, instead of the actual object. hence removed as it won't help in debugging